include/shark/ObjectiveFunctions/Benchmarks/MultiObjectiveBenchmark.h

```diff
@@ -0,0 +1,189 @@
+	template<int N> struct generate_integer_sequence: integer_sequence_concat<typename generate_integer_sequence<N/2>::type, typename generate_integer_sequence<N-N/2>::type>::type {};
+	template <> struct generate_integer_sequence<0>: integer_sequence<>{};
+	template <> struct generate_integer_sequence<1>: integer_sequence<0>{};
+};
+
+/// \brief Creates  a multi-objective Benchmark from a set of given single objective functions
+///
+/// A variadic template is used to generate a set of benchmarks.
+/// eg MultiObjectiveBenchmark<Sphere,Ellispoid,Rosenbrock> sets up a three-objective Benchmark.
+///
+/// A random rotation and translation is applied to each benchmark function, thus
+/// MultiObjectiveBenchmark<Sphere,Sphere> forms a non-degenerate front.
+/// the ith objective can be queried via the get<i> member function.
+///
+/// The generated translations are approximately sampled from the unit ball and starting points are also drawn
+/// by the same distribution around a random optimum (assuming the optimum is at (0,0) of the untranslated function
+template<class ... Objectives>
+class MultiObjectiveBenchmark: public MultiObjectiveFunction{
+public:
+	MultiObjectiveBenchmark(std::size_t numVariables = 5){
+		m_features |= CAN_PROPOSE_STARTING_POINT;
+		m_features |= HAS_FIRST_DERIVATIVE;
+		setupRotations(typename detail::generate_integer_sequence<sizeof...(Objectives)>::type());
+		setNumberOfVariables(numVariables);//prevent that different objectives have different default number of variables.
+		for(auto& f: m_rotations){
+			//if one function does not have a first derivative, no function has
+			if(!f.hasFirstDerivative())
+				m_features.reset(HAS_FIRST_DERIVATIVE);
+		}
+	};
+	
+	///\brief Name of the Benchmark
+	///
+	/// The name has the form Objective1/Objective2/Objective3/.../ObjectiveN
+	/// where ObjectiveK is the name of the k-th objective.
+	std::string name()const{
+		return generateName(typename detail::generate_integer_sequence<sizeof...(Objectives)>::type());
+		
+	}
+	
+	bool hasScalableDimensionality()const{
+		return true;
+	}
+	
+	void setNumberOfVariables( std::size_t numberOfVariables ){
+		for(auto& f: m_rotations){
+			f.setNumberOfVariables(numberOfVariables);
+		}
+	}
+	
+	std::size_t numberOfObjectives()const{
+		return m_rotations.size();
+	}
+	
+	std::size_t numberOfVariables()const{
+		return get<0>().numberOfVariables();
+	}
+	
+	template<int N>
+	typename std::tuple_element<N, std::tuple<Objectives...> >::type& get(){
+		return std::get<N>(m_objectives);
+	}
+	template<int N>
+	typename std::tuple_element<N, std::tuple<Objectives...> >::type const& get()const{
+		return std::get<N>(m_objectives);
+	}
+	
+	///\ Initializes the functions as well as picks random rotations and translations
+	void init() {
+		m_translations.clear();
+		
+		for(auto& f: m_rotations)
+		{
+			RealVector translation(numberOfVariables());
+			for(double& v: translation){
+				v=Rng::gauss(0,1)/numberOfVariables();
```

Comment:
Shouldn't this divide by sqrt(numberOfVariables()) to obtain an expected squared norm of one?